### ISSUE TITLE
Azure AI Projects: Async function `send_email_async` is not called

### ISSUE THREAD

**Issue by jhakulin** [Azure/azure-sdk-for-python#38898]
- **Package Name**: azure-ai-projects
- **Package Version**: 1.0.0b3
- **Operating System**: Windows
- **Python Version**:  3.12.8

**Describe the bug**
When I run this sample https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_stream_eventhandler_with_functions_async.py, the `send_email_async` is not called.

**To Reproduce**
1.  Run the sample

**Expected behavior**
`send_email_async` will be called as that is needed for user query in the sample

**Screenshots**

**Additional context**
The problem can be avoided if I add following docstrings to the `send_email_async` method

```python
async def send_email_async(recipient: str, subject: str, body: str) -> str:
    """
    Sends an email with the specified subject and body to the recipient.

    :param recipient (str): Email address of the recipient.
    :param subject (str): Subject of the email.
    :param body (str): Body content of the email.
    :return: Confirmation message.
    :rtype: str
    """
    await asyncio.sleep(1)
    return send_email(recipient, subject, body)
```
After that agent will know more details about the function and knows to call it successfully.

**Comment by jhakulin** [comment]
This will be fixed in 1.0.0b4 version soon.

**Comment by jhakulin** [comment]
released in https://pypi.org/project/azure-ai-projects/1.0.0b4/
